threescale/src/utils.rs

```diff
@@ -1,13 +1,58 @@
-use crate::structs::Application;
+use crate::structs::{ Application, PeriodWindow };
+use proxy_wasm::hostcalls::{ set_shared_data, get_shared_data};
+use log::info;
+use std::time::{ SystemTime, Duration };
 
 // Returns Application from shared data with CAS integer
 pub fn get_application_from_cache(key: &str) -> Option<(Application,u32)> {
-    None
+    match get_shared_data(&key).unwrap()
```

Comment:
I'd prefer that when `get_shared_data()` fails we not only panic but also we have an explicit path in the code panicking, such as:
```rust
match get_shared_data(&key) {
    Ok(Some(bytes), cas) => { ... // deal with happy path },
    Ok(None, cas) => { ... // handle no data for this key - perhaps panic for the time being },
    Err(status) => { ... // handle not found for this key - perhaps panic for the time being },
}
```

In general most, but not all, `unwrap()` calls mean that there is some match arm or else clause that needs to be handled (ie. we want to avoid panics), so having them explicitly show up in the code, even if just panicking, helps people having them fixed earlier than otherwise.

Reply:
Thanks for the great feedback! I'll keep them in mind and add the workflow to automate as mentioned by you.